Assets/Scripts/Units/Enemy.cs

```diff
@@ -7,7 +7,46 @@ namespace RogueApeStudio.Crusader.Units
 {
     public abstract class Enemy : MonoBehaviour
     {
+        [Header("Movement settings")]
+        [SerializeField] protected float _stopDistance = 1.8f;
+        [SerializeField] protected float _startAttackDistance = 2.0f;
+        [Header("Combat settings")]
+        [SerializeField] protected float _selfStunDuration = 1;
+        [SerializeField] protected float _detectionRange = 20;
+        [SerializeField] protected float _damageAmount = 5;
+        [SerializeField] protected float _attackSpeed = 1;
+        [Header("Death settings")]
+        [SerializeField] protected float _destroyTime = 3;
+        [Header("Dependancies")]
+        [SerializeField] protected UnitMovement _localUnitMovement;
+        [SerializeField] protected Animator _localAnimator;
+        [SerializeField] protected Health _health;
+        [SerializeField] protected Collider _collider;
+
+        public UnitMovement LocalUnitMovement => _localUnitMovement;
+        public Animator LocalAnimator => _localAnimator;
+        public float StartAttackDistance => _startAttackDistance;
+        public float DestroyTime => _destroyTime;
+        public float SelfStunDuration => _selfStunDuration;
+        public float DetectionRange => _detectionRange;
+
         public abstract bool Stun();
         public abstract bool IsDead();
+
+        public void DestroySelf()
```

Comment:
Isn't this a job for the `EnemyHealth` script? Why does the pathfinding/combat system need to destroy itself?

Reply:
Its an enemy behaviour script, the health script does not contain any functionality for death.

Reply:
Sure, I can see that.
I'd have preferred to see it in `EnemyHealth`, but sure.